### PR TITLE
Implement OP_WITHIN opcode 

### PR DESCRIPTION
**Pull request by supreme2580**
- [x] issue # https://github.com/keep-starknet-strange/shinigami/issues/19
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

I implemented the code for op_within

### CODE REVIEW

File: src/opcodes/opcodes.cairo [hunk 1]
```diff
@@ -174,6 +194,27 @@ pub mod Opcode {
...
+        let b = engine.dstack.pop_int();
+        engine.dstack.push_int(if a > b {
+            0
+        } else {
+            1
+        });
+    }
+
+    fn opcode_within(ref engine: Engine) {
+        let a = engine.dstack.pop_int();
```
Comment (b-j-roberts) [inline src/opcodes/opcodes.cairo:208]: Would be prefered to make the variable naming here more explicit, like min_value, max_value, ...

File: src/opcodes/opcodes.cairo [hunk 2]
```diff
@@ -174,6 +194,27 @@ pub mod Opcode {
...
+        let b = engine.dstack.pop_int();
+        engine.dstack.push_int(if a > b {
+            0
+        } else {
+            1
+        });
+    }
+
+    fn opcode_within(ref engine: Engine) {
```
Comment (b-j-roberts) [inline src/opcodes/opcodes.cairo:207]: Please refer to this implementation to see the proper ordering of variables on the stack : https://github.com/btcsuite/btcd/blob/b161cd6a199b4e35acec66afc5aad221f05fe1e3/txscript/opcode.go#L1844

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: src/opcodes/opcodes.cairo (hunk 1 above)
Comment (supreme2580) [reply src/opcodes/opcodes.cairo:208]: Will rename the variables now

File: src/opcodes/opcodes.cairo (hunk 2 above)
Comment (supreme2580) [reply src/opcodes/opcodes.cairo:207]: Working on this

File: src/opcodes/opcodes.cairo (hunk 1 above)
Comment (supreme2580) [reply src/opcodes/opcodes.cairo]: Hi I just pushed an update to rename the variables to be more explicit

File: src/opcodes/opcodes.cairo (hunk 2 above)
Comment (supreme2580) [reply src/opcodes/opcodes.cairo:207]: Hi, opcode_within is number 165 in the order, please could you tell me what you mean? I have pushed an update where I renamed and reordered the variables though

Comment (b-j-roberts) [reply src/opcodes/opcodes.cairo:207]: So we have to make sure the values poped off the stack are in the same order as the bitcoin spec, so here `max_value` should be popped first, then `min_value`, then `selected_value`

Comment (supreme2580) [reply src/opcodes/opcodes.cairo]: Screenshot 2024-07-26 at 17 06 37

Hi @b-j-roberts I have rearranged the variables based on the order max before min before value

**Pull request closed without merging**